### PR TITLE
#7: Set up automatic linting using ruff

### LINKED ISSUE #7: Set up automatic linting using Ruff
## Problem Statement

Ruff is currently installed with the UV environment, but it is not running automatically. 

## Proposed Solution

1. Create a pre-commit job that lints before commit.
2. Create a workflow that ensures there are no linting errors before commit

### PR DESCRIPTION
**Pull request by peterkaloyannis**
## Issue

Closes #7

## Description

Adds pre-commit and CI to lint the code and make sure its all happy! 

## Verification

Job passes here even though I ran rough over the entire repo to manually relint everything.

### CODE REVIEW

File: feed_me/src/feed_me/cli.py
Comment (peterkaloyannis) [inline feed_me/src/feed_me/cli.py:1]: This change was automatically done by ruff.

File: feed_me/src/feed_me/task_manager.py
Comment (peterkaloyannis) [inline feed_me/src/feed_me/task_manager.py:1]: This change was done by ruff

File: feed_me/tests/test_task_manager.py
Comment (peterkaloyannis) [inline feed_me/tests/test_task_manager.py:1]: Ruff actually forced me to make this code improvement for the test to pass. Good job little robot

File: .github/workflows/ruff.yml
Comment (peterkaloyannis) [inline .github/workflows/ruff.yml:1]: This workflow file is created by the creators for ruff.

File: .pre-commit-config.yaml
Comment (peterkaloyannis) [inline .pre-commit-config.yaml:1]: This is a pre-commit file. I will show you what this does in person, but it will help keep our repo clean!

**Review by mnavarrocastillo — APPROVED** [review]

**Pull request merged**